core/src/main/java/org/apache/iceberg/RESTPlanningMode.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import java.util.Locale;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+public enum RESTPlanningMode {
+  REQUIRED("required"),
+  SUPPORTED("supported"),
+  UNSUPPORTED("unsupported");
```

Comment:
I think thats also fair if we need to have an additional `unsupported` flag, i think not having it also signals the same thing.

Reply:
Will let others chime in here though before making a revision.

Reply:
Sounds good. After some more thought, my 2c is that I think I'd rather us try and get the model + client side changes into 1.7 rather than expand the scope with this aspect since it's quite useful without these things defined. I'd rather not have client side changes depend on a spec change decision.

For now, I think keeping it simple with a catalog client side property for controlling if server side planning is performed is the way.

Once the model and client side changes are in, I think it'd make total sense to revisit the spec changes you mentioned.

cc @rdblue @danielcweeks for their thoughts.

Reply:
cc @jackye1995